unittest/gunit/thread_utils.h

```diff
@@ -115,4 +115,22 @@ class Notification
 
 }  // namespace thread
 
+extern void disable_core_dumps(void);
+extern void restore_core_dump_rlimit(void);
+
+
+#define MY_EXPECT_DEATH_IF_SUPPORTED(statement, regex)  \
+  do {                                                  \
+    disable_core_dumps();                               \
```

Comment:
May be a RAII capsule which would store old and set new value in the constructor and restore in the destructor?
Something like the following with proper error checking. No static initialization at all here.
```c++
class disable_core_dumps_capsule
{
  public:
    disable_core_dumps_capsule(): captured_core_rlimit_()
    {
      ... =getrlimit(RLIMIT_CORE, &captured_core_rlimit_);
      struct rlimit disabled_core_rlimit= captured_core_rlimit_;
      disabled_core_rlimit.rlim_cur= 0;
      ...= setrlimit(RLIMIT_CORE, &disabled_core_rlimit);
    }
    ~disable_core_dumps_capsule()
    {
      ...= setrlimit(RLIMIT_CORE, &captured_core_rlimit_);
    }
  private:
    disable_core_dumps_capsule(const disable_core_dumps_capsule&);
    disable_core_dumps_capsule& operator = (const disable_core_dumps_capsule&);

    struct rlimit captured_core_rlimit_;
};
```